ghcjs-src/Miso/Html/Internal.hs

```diff
@@ -72,10 +72,14 @@ import           Miso.Event.Types
 import           Miso.String
 import           Miso.FFI
 
--- | Type def for constructing event subscriptions
-type Sub a m = IO m -> (a -> IO ()) -> IO ()
+-- | Type synonym for constructing event subscriptions.
+--
+-- The first argument passed to a subscription provides a way to
+-- access the current value of the model. The callback is used to
```

Comment:
Might be worthwhile to add `".. access the current value of the model in a non-blocking fashion"`, or something to ensure the user it's a non-blocking read.